### ISSUE TITLE
Using flows for density estimation

### ISSUE THREAD

**Issue by pratham16cse** [zalandoresearch/pytorch-ts#103]
Thanks for providing time-series compatible normalizing flows in [flows](https://github.com/zalandoresearch/pytorch-ts/blob/62a18cfd0f19fc2247fc7e79e77f44b8dab6c4c9/pts/modules/flows.py) file. I am interested in using normalizing flows for density estimation. 

I have a couple of doubts about how masks are created in MADE block of MAF:
Suppose the input is of shape `B x T x D` where `B = batch size`, `T = Sequence length` and `D = Dimension of the sequence` and I want to estimate the density at each position (Eq. 2 of [MAF Paper](https://papers.nips.cc/paper/2017/file/6c1da886822c67822bcf3679d04369fa-Paper.pdf)). In this case,

1. What should be the `input_size` to [MAF](https://github.com/zalandoresearch/pytorch-ts/blob/62a18cfd0f19fc2247fc7e79e77f44b8dab6c4c9/pts/modules/flows.py#L381) network?
2. Can you give a toy example of what masks will be generated in the MADE block?
3. Also, how the `cond_label_size` would fit in the whole equation?

Perhaps specifying the shapes of the input and output of each block in flows file would be helpful. Please let me know if more elaboration on the setup is needed.

Thanks,
- Prathamesh

**Comment by kashif** [comment]
@pratham16cse i would not use these specific modules for general density estimation as they are specific to the conditional estimation of a sequence of multivariate vectors...

I do not remember exactly the variables names right now... I would need to look at the code again...

I believe the flows because they are invertiable do not change the size of the input so all that remains the same... so i belive the input_size might be multivariate dim size and cond_lable size might be the size of the condition vector... I have to check

**Comment by kashif** [comment]
yes lets have a video chat soon and discuss things! I am interested in your work and wish to also understand it further. Kindly send me an invite at kashif.rasul@gmail.com for some reasonable time... I would love to chat!
